tests/integration/targets/mso_schema_site_contract_service_graph/tasks/main.yml

```diff
@@ -0,0 +1,770 @@
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    service_graph_schema: ansible_schema
+    service_graph_template: ansible_template1
+    service_graph: ansible_service_graph
+    node_relationship:
+      - cluster_interface_device: ansible_tenant_firewall1
+        provider_connector_cluster_interface: clu_if1
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if1
+        consumer_connector_redirect_policy: redirect_policy1
+      - cluster_interface_device: ansible_tenant_adc
+        provider_connector_cluster_interface: clu_if3
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if3
+        consumer_connector_redirect_policy: redirect_policy1
+        consumer_subnet_ips: ["1.1.1.1/24", "4.4.4.4/24"]
+      - cluster_interface_device: ansible_tenant_other
+        provider_connector_cluster_interface: clu_if4
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if4
+        consumer_connector_redirect_policy: redirect_policy1
+    state: present
+  check_mode: true
+  ignore_errors: true
+  register: add_sg_cm
+
+- name: Bind the Contract1 with site service graph - ansible_service_graph - normal_mode
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *add_site_con_sg_cm
+  ignore_errors: true
+  register: add_sg_nm
+
+- name: Bind the Contract1 with site service graph - ansible_service_graph - normal_mode - again
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *add_site_con_sg_cm
+  ignore_errors: true
+  register: add_sg_nm_again
+
+- name: Assertions check for the bind site contract service graph
+  assert:
+    that:
+      - add_sg_cm is changed
+      - add_sg_cm.current != {}
+      - add_sg_cm.previous == {}
+      - add_sg_nm is changed
+      - add_sg_nm.current != {}
+      - add_sg_nm.current.serviceGraphRef.serviceGraphName == "ansible_service_graph"
+      - add_sg_nm.current.serviceGraphRef.templateName == "ansible_template1"
+      - add_sg_nm.current.serviceNodesRelationship.1.consumerConnector.subnets | length == 2
+      - add_sg_nm.current.serviceNodesRelationship.2.consumerConnector.clusterInterface.dn == "uni/tn-ansible_tenant/lDevVip-ansible_tenant_other/lIf-clu_if4"
+      - add_sg_nm.current.serviceNodesRelationship.2.consumerConnector.redirectPolicy.dn == "uni/tn-ansible_tenant/svcCont/svcRedirectPol-redirect_policy1"
+      - add_sg_nm.current.serviceNodesRelationship.2.providerConnector.clusterInterface.dn == "uni/tn-ansible_tenant/lDevVip-ansible_tenant_other/lIf-clu_if4"
+      - add_sg_nm.current.serviceNodesRelationship.2.providerConnector.redirectPolicy.dn == "uni/tn-ansible_tenant/svcCont/svcRedirectPol-redirect_policy1"
+      - add_sg_nm.previous == {}
+      - add_sg_nm_again is not changed
+      - add_sg_nm_again.current != {}
```

Comment:
Should we check the individual attributes of current and previous? Applies to all tests assertions